apps/meteor/server/services/omnichannel-verification/service.ts

```diff
@@ -63,30 +65,37 @@ export class OmnichannelVerification extends ServiceClassInternal implements IOm
 		});
 	}
 
-	private async wrongInput(room: IOmnichannelGenericRoom, _isWrongOTP: boolean): Promise<boolean> {
+	private async wrongInput(room: IOmnichannelRoom, _isWrongOTP: boolean): Promise<boolean> {
 		const limitWrongAttempts: number = settings?.get('Livechat_LimitWrongAttempts') ?? 3;
 		if (!room?.wrongMessageCount) {
 			await LivechatRooms.updateWrongMessageCount(room._id, 1);
-		} else if (room.wrongMessageCount + 1 >= limitWrongAttempts) {
+			return true;
+		}
+		if (room.wrongMessageCount + 1 >= limitWrongAttempts) {
+			const { emailCode } = room.services || {};
 			await Promise.all([
 				LivechatRooms.updateWrongMessageCount(room._id, 0),
 				LivechatRooms.updateVerificationStatusById(room._id, RoomVerificationState.verifiedFalse),
+				emailCode && _isWrongOTP
+					? Promise.all(emailCode.map(({ code }) => LivechatRooms.removeEmailCodeByRoomIdAndCode(room._id, code)))
+					: Promise.resolve(),
+				// eslint-disable-next-line no-constant-condition
+				room?.servedBy?._id || 'rocket.cat'
+					? sendMessage(
```

Comment:
This should be outside promise.all since we should only send the confirmation message once all the operations in these promise is a success.